meetings/2024-04/april-08.md

```diff
@@ -0,0 +1,1049 @@
+
+PFC: Okay, thanks. That’s good information as well.
+
+PFC: All right. (Slide 9) I would like to request consensus on this pull request linked here that fixes the rounding bug that I described.
+
+USA: All right. Let’s give it a minute. I would also reiterate that any sort of statements of explicit support are also welcome. All right. There doesn’t seem to be anything in the queue. So you have consensus.
+
+PFC: Okay. Thanks. (Slide 10) I took the liberty of writing a proposed summary for the note, which I will show here and paste into the notes.
+
+### Speaker's Summary of Key Points
+
+- Consensus was reached on a normative change to fix a bug in rounding that occurred in rare cases having to do with DST.
+- Over the next few weeks, we plan to dig into remaining concerns from TC39 delegates, particularly with the goal of reducing complexity.
+- Follow the checklist in [#2628](https://github.com/tc39/proposal-temporal/issues/2628) for updates.
+
+## Duplicate named capture groups for stage 4
+
+Presenter: Kevin Gibbons (KG)
+
+- [PR](https://github.com/tc39/ecma262/pull/2721)
+
+KG: Okay. So duplicate named capture groups. As a reminder, since it’s been a couple of years since I presented this, this is a feature that allows you to have the same capturing group name into two parts of a regular expression, with the constraint that they can’t both participate in the match, which is to say they have to be in different alternatives. So separated by a pipe. But otherwise, it works exactly like you would expect. You can use back references to the capturing group name. The.groups object of the result will contain the value from whichever one actually matched. In the case of repetition groups the last reputation group defines it, the same way it works for regular capturing groups.
+
+KG: The specification text is quite simple, although it’s not been reviewed by all of the other editors yet. That is, it’s not been reviewed as a pull request. But of course the specification text is approved as part of getting to Stage 3. It has been shipping in Safari for a while, and shipping in Chrome in 125, which isn’t stable yet. I believe it’s currently in the dev channel. Since this is only causing syntax to become legal which wasn’t previously legal, there isn’t much risk of web incapability. SpiderMonkey uses V8's RegExp engine as the underlying thing for their engine. They have to do a little bit of integration work to expose the new functionality. And I believe that work is underway. So it’s not yet shipping in Firefox.
```

Comment:
```suggestion
KG: The specification text is quite simple, although it’s not been reviewed by all of the other editors yet. That is, it’s not been reviewed as a pull request. But of course the specification text is approved as part of getting to Stage 3. It has been shipping in Safari for a while, and shipping in Chrome in 125, which isn’t stable yet. I believe it’s currently in the dev channel. Since this is only causing syntax to become legal which wasn’t previously legal, there isn’t much risk of web incompatibility. SpiderMonkey uses V8's RegExp engine as the underlying thing for their engine. They have to do a little bit of integration work to expose the new functionality. And I believe that work is underway. So it’s not yet shipping in Firefox.
```